scripts/components/cart.jsx

```diff
@@ -0,0 +1,504 @@
+			error = data.error;
+		}
+		return {
+			applied: data.valid === true && data.code && data.code !== '' && !error,
+			code: !error ? (data.code || '').toUpperCase() : '',
+			isAuto: !!data.isAuto,
+			amount: data.discount < 0 ? (data.discount * -1) : (data.discount || 0),
+			error,
+		};
+	}
+
+	/* -------------------
+		Actions
+	------------------- */
+
+	onChangeQuantity = (item, qty, callback) => {
+		this.setState({ isLastStockKey: '' }, () => {
+			snCart.changeQuantity(item, qty, (newQty) => {
+				this.setState({ isLastStockKey: item.key });
+				callback(newQty);
+			});
+		});
+	}
+
+	onChangeVariant = (item, newVariantId) => {
+		if (item.id !== newVariantId) {
+			snCart.replaceItem(item.key, newVariantId, item.quantity);
+		}
+	}
+
+	onRemoveItem = (item) => {
+		snCart.removeItem(item.id);
+	}
+
+	onAddUpsell = (upsell) => {
+		if (upsell.upgrade_bundle_method === 'replace') {
+			snCart.replaceItem(upsell.targetId, upsell.replaceToId, 1);
+		} else {
+			snCart.addItem(upsell.replaceToId, 1);
+		}
+	}
+
+	onApplyDiscountCode = (code) => {
+		this.setState({ loadingDiscount: true }, () => {
+			snCart.applyDiscountCode(code).then((discountData) => {
+				if (discountData.enabled === false || discountData.isValid === false) {
+					this.setState({
+						loadingDiscount: false,
+						discountData: this.getDiscountDataDisplay({ reason: discountData.error }),
+					});
+				} else {
+					this.setState({
+						loadingDiscount: false,
+						discountData: this.getDiscountDataDisplay({
+							...discountData,
+							valid: true,
+							error: '',
+						}),
+					});
+				}
+			});
+		});
+	}
+
+	onRemoveDiscountCode = () => {
+		this.setState({ loadingDiscount: true }, () => {
+			snCart.removeDiscountCode();
+			this.setState({
+				loadingDiscount: false,
+				discountData: this.getDiscountDataDisplay(),
+			});
+		});
+	}
+
+	onToggleManualGwp = (id) => {
+		this.setState({
+			loadingManualGwp: { loading: true, id },
+		}, () => {
+			const newSelected = snCart.toggleManualGwp(this.state.manualGwp, id);
+			this.setState((prevState) => ({
+				manualGwp: {
+					...prevState.manualGwp,
+					selectedKey: newSelected,
+				},
+				loadingManualGwp: { loading: false, id: -1 },
+			}));
+		});
+	}
+
+	handleKeyDown = (e) => {
+		if (e.keyCode === 13) {
+			e.preventDefault();
+		}
+	}
+
+	submitForm() {
+		if (this.formRef) {
+			this.formRef.submit();
+		}
+	}
+
+	render() {
+		const {
+			cart,
+			loadingInit,
+			isLastStockKey,
+			itemCount,
+			totalPrice,
+			loadingDiscount,
+			discountData,
+			manualGwp,
+			loadingManualGwp,
+			upsellData,
+			shippingData,
+			discountMeter,
+		} = this.state;
+		return (
+			<div className="modal-dialog modal-md m-0 w-100 float-right">
+				<div className="modal-content vh-100 border-0 rounded-0">
+					<div className="modal-body mobile-wrapper pt-0 px-lg-0">
+						<div className="container d-flex flex-column align-items-stretch text-center pt-2">
```

Comment:
do we need all of these DIVs? Can we try to reduce their number?

Reply:
bootstrap use the animation css on .modal-dialog
.modal-content prevent the popup to close when anything is clicked
I tried to remove the .modal-body, but then somehow I cannot get the modal-footer to be fixed at the bottom, position sticky and absolute doesn't work

As for the container, its a wrapper for the header part of the cart, needed to get the gutter padding